openmdao/core/group.py

```diff
@@ -4030,13 +4026,15 @@ def _declared_partials_iter(self):
 
     def _get_missing_partials(self, missing):
         """
-        Return a list of (of, wrt) tuples for which derivatives have not been declared.
+        Provide (of, wrt) tuples for which derivatives have not been declared in the system.
 
         Parameters
         ----------
         missing : dict
             Dictionary containing list of missing derivatives keyed by system pathname.
         """
+        if self._has_approx:
```

Comment:
Shouldn't we also do this same sort of thing in Component._get_missing_partials?

Reply:
For components, I think _has_approx is too broad of a brush, since it is true if only some partials are approximated. The get_missing_partials method should account for this.